hook/hook.go

```diff
@@ -240,10 +240,27 @@ func ExecuteWithRunCapture(log log15.Logger,
 }
 
 func runArgsFactory(log log15.Logger, config *conf.Config, workingDir string) []string {
+	var mountedVolume, volumeFlag string
+	mountedVolume = "/repo_root"
+	volumeFlag = os.Getenv(constants.EnvVolumeFlag)
+
+	// Adding volume flag to either allow read only or to maintain same volume flags
+	switch volumeFlag {
+	case "z", "ro":
+	default:
+		log.Info("Setting the default volume flag as empty.")
+		volumeFlag = ""
```

Comment:
This is the wrong validation. If the value was set we need to ensure it's one of the valid values or reset it to `""`. This should read

```golang
case "", "z", "ro":
default:
    log.Warn("invalid volume flag set. resetting to \"\"", "volumeFlag", volumeFlag)
    volumeFlag = ""
```

alternatively we change the signature of this function to return `(runArgs []string, success bool)`